### PR TITLE
Added Guess Handling

### PR DESCRIPTION
**Pull request by dbolusan**
Implementing Guess Handling for guesses made by user in comparison to challenge words.

### CODE REVIEW

File: lexic-frontend/src/checkGuess.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// List of Statuses
+// 0 : Letter is not in the word/ Initial State
+// 1 : Letter is in the word, not in right index
+// 2 : Letter is in the word, correct index
```
Comment (WJKw0k) [inline lexic-frontend/src/checkGuess.js:4]: Might want to add a enum or something instead of magic numbers - but not a big deal

**Review by WJKw0k — APPROVED** [review]

File: lexic-frontend/src/checkGuess.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  };
+  return dict;
+}
+
+// helper -- inserts '_' when letter is used
+function insertLetter(str, i) {
+  return `${str.substring(0, i)}_${str.substring(i + 1)}`;
+}
+
+// returns a list of status, reals, on status of correct letters
```
Comment (Kanaz312) [inline lexic-frontend/src/checkGuess.js:33]: return comment is confusing. Not sure if the return is a list of status and reals or what. Also not sure what "on status of correct letters" means.

**Review by Kanaz312 — CHANGES_REQUESTED** [review]
Return comments need some changes, but otherwise looks really nice!
Also nice job putting return comments

File: lexic-frontend/src/checkGuess.js (hunk 2 above)
Comment (dbolusan) [reply lexic-frontend/src/checkGuess.js]: Sounds good! I rephrase the return comments to indicate that these are a list of integers that represents letters

**Review by Kanaz312 — APPROVED** [review]
Looks good!

**Review by WJKw0k — APPROVED** [review]
Looks good to me. Thanks for documenting the code!

**Review by Tylertyca — APPROVED** [review]
Looks good. Nice tests.

**Pull request merged**